backend/test/integration/GameStartEvents.integration.test.js

```diff
@@ -0,0 +1,85 @@
+const Client = require('socket.io-client');
+const SocketIOServer = require('../../index');
+const config = require('../../config.json');
+const CreateLobbyDTO = require('../../domain/DTO/request/CreateLobbyDTO');
+const JoinLobbyDTO = require('../../domain/DTO/request/JoinLobbyDTO');
+
+describe('GameStartEvents integration tests', () => {
+    const port = process.env.PORT || config.local_port;
+
+    let clientSockets = [];
+
+    // Create a new client, and connect it to the server via a socket
+    beforeEach(async (done) => {
+        // Host creates lobby
+        async function connectAndCreateLobby() {
+            return new Promise((resolve) => {
+                clientSockets[0] = new Client(`http://localhost:` + port);
+                clientSockets[0].on('connect', () => {
+                    clientSockets[0].emit('create-lobby', new CreateLobbyDTO('Leon'));
+                    clientSockets[0].on('lobby-code', (createLobbyDTO) => {
+                        resolve(createLobbyDTO.code);
+                    });
+                });
+            });
+        }
+
+        lobbyCode = await connectAndCreateLobby();
+        done();
+    });
+
+    // Disconnect each socket connected to the server
+    afterEach((done) => {
+        const sockets = SocketIOServer.io.sockets.sockets;
+        sockets.forEach(function (socket, key) {
+            socket.disconnect(true);
+        });
+        done();
+    });
+
+    // Close the server once all tests are done
+    afterAll(() => {
+        SocketIOServer.server.close();
+    });
+
+    test('integration test start-game 6 players', async (done) => {
+        // Connect to lobby created in before each with 5 other players
+        async function connectAndJoin(index) {
+            return new Promise((resolve) => {
+                clientSockets[index] = new Client(`http://localhost:` + port);
+                clientSockets[index].on('connect', () => {
+                    clientSockets[index].emit('join-lobby', new JoinLobbyDTO('Leon' + index.toString(), lobbyCode));
```

Comment:
I think this might cause issues when we start enforcing unique nicknames. Maybe use the index in the name string to ensure that each socket that joins has a unique nickname?

Reply:
thats what i did

Reply:
oh i'm blind lmao